### PR TITLE
Add docs page for using smart IDs in API v2

### LINKED ISSUE #317: Add docs for using smart IDs in API requests
vanilla/vanilla#7175 added support for smart IDs. They allow for things like transparently looking up a category ID from a URL code or getting a user ID based on an SSO foreign ID. They're great. But we don't have docs for using them and they aren't the most intuitive things to use. We need documentation for using smart IDs in API requests.

### PR DESCRIPTION
**Pull request by initvector**
Here, we're adding documentation on using smart IDs in API v2.

Closes #317

### CODE REVIEW

File: content/help/apiv2/smart-ids.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: Smart IDs
+tags:
+- API
```
Comment (charrondev) [inline content/help/apiv2/smart-ids.md]: Could we add a tag here "ForeignID", "Foreign" as well?
```suggestion
- API
- ForeignID
- Foreign
```

File: content/help/apiv2/smart-ids.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- APIv2
+- API v2
+category: "apiv2"
+swagger: true
```
Comment (charrondev) [inline content/help/apiv2/smart-ids.md]: I don't think this is supposed to be here. It adds the swagger js to the page.
```suggestion
```

Comment (initvector) [reply content/help/apiv2/smart-ids.md]: > It adds the swagger js to the page.

TIL

**Comment by initvector** [comment]
Blurb about failure conditions has been added 👍

**Pull request merged**